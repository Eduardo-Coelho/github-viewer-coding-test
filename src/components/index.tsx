export * from './commitDetails/CommitDetails';
export * from './commitTable/CommitTable';
export * from './repoSearch/RepoSearch';
export * from './loader/Loader';
export * from './modal/Modal';
