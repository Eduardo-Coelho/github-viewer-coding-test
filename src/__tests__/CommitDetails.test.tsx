import React from 'react';
import { render } from '@testing-library/react';
import { CommitDetails } from '../components';

test('CommitDetails matches snapshot', () => {
  const mockCommit = {
    commit: {
      author: { name: 'John Doe' },
      committer: { date: '2023-05-21T12:34:56Z' },
      message: 'Initial commit',
      url: 'http://example.com/commit/1',
    },
  };

  const { asFragment } = render(<CommitDetails commit={mockCommit as any} />);
  expect(asFragment()).toMatchSnapshot();
});

test('CommitDetails matches snapshot when commit is null', () => {
  const { asFragment } = render(<CommitDetails commit={null} />);
  expect(asFragment()).toMatchSnapshot();
});
