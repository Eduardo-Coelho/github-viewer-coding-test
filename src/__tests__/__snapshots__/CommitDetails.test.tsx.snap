// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitDetails matches snapshot 1`] = `
<DocumentFragment>
  <div
    test-id="commit-details"
  >
    <h3>
      Commit Details:
    </h3>
    <p>
      <strong>
        Author:
      </strong>
       John Doe (John Doe)
    </p>
    <p>
      <strong>
        Date:
      </strong>
       21/05/2023, 13:34:56
    </p>
    <p>
      <strong>
        Message:
      </strong>
       Initial commit
    </p>
    <p>
      <a
        href="http://example.com/commit/1"
        rel="noreferrer"
        target="_blank"
      >
        View Commit
      </a>
    </p>
  </div>
</DocumentFragment>
`;

exports[`CommitDetails matches snapshot when commit is null 1`] = `
<DocumentFragment>
  <div
    test-id="commit-details"
  >
    <h3>
      Commit Details:
    </h3>
    <p>
      <strong>
        Author:
      </strong>
       N/A (N/A)
    </p>
    <p>
      <strong>
        Date:
      </strong>
       N/A
    </p>
    <p>
      <strong>
        Message:
      </strong>
       N/A
    </p>
    <p>
      <a
        rel="noreferrer"
        target="_blank"
      >
        View Commit
      </a>
    </p>
  </div>
</DocumentFragment>
`;
